src/main/scala/com/microsoft/hyperspace/logging/GenericHyperspaceLogging.scala

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (2020) The Hyperspace Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.microsoft.hyperspace.logging
+
+import scala.util.{Success, Try}
+
+import org.apache.spark.internal.Logging
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.util.hyperspace.Utils
+
+import com.microsoft.hyperspace.logging.Constants._
+
+trait HyperspaceLogging extends Logging {
+  def logEvent(event: HyperspaceEvent): Unit = HyperspaceLogging.logger.logEvent(event)
+}
+
+object HyperspaceLogging extends Logging {
+  private val logger: HyperspaceLogging = {
+    val spark = SparkSession.getActiveSession
+    val className = if (spark.isDefined) {
+      spark.get.conf.get(HYPERSPACE_LOGGER_CLASS, DEFAULT_HYPERSPACE_LOGGER_CLASS)
+    } else {
+      logDebug(s"$HYPERSPACE_LOGGER_CLASS is not set in spark conf, default logger used.")
+      Constants.DEFAULT_HYPERSPACE_LOGGER_CLASS
+    }
+
+    Try(Utils.classForName(className).newInstance) match {
+      case Success(logger: HyperspaceLogging) => logger
+      case _ =>
+        logDebug(s"Hyperspace logger is not appropriately set, using default logger.")
+        Utils
+          .classForName(DEFAULT_HYPERSPACE_LOGGER_CLASS)
+          .newInstance()
+          .asInstanceOf[HyperspaceLogging]
+    }
+  }
+}
+
+/**
+ * Logger that logs Hyperspace events to spark logs.
+ */
+class GenericHyperspaceLogging extends HyperspaceLogging {
+  override def logEvent(event: HyperspaceEvent): Unit = {
+    logInfo(event.dimensions.mkString("; "))
```

Comment:
I think no op should be fine.

Reply:
Thanks @imback82 , 
I added logInfo so that if nothing is provided, we will see the events at least in spark application logs. With no-op, the events won't be seen. Is that the expected behavior?

If we go with no-op, this will be the behavior:
- for generic logging (info, error, warn, debug etc), use `HyperspaceLogging.logInfo(..), logError(...)` which logs to spark application logs
- for event logging, use `HyperspaceLogging.logEvent` which would be a no-op if no external HyperspaceLogging implementation is provided

If this is ok, I can make this change. Thanks

Reply:
yes. if the user wants to capture the event, they can provide the event logger. btw, do we want `HyperspaceLogging` to extend `Logging`? or should we decouple it  as a `HyperspaceEventLogging` so that you can do `class A extends Logging with HyperspaceEventLoggin`?

Reply:
cc: @AFFogarty with event logging backgound

Reply:
Thanks @imback82 , @AFFogarty 
Here's the summary from both your comments:
1. Let's keep logging (info/error etc) and emitting events separate
2. Logging will be handled by spark.internal.Logging class. We don't need to implement this logging or add a redirection to this logging through HyperspaceLogging
3. Hyperspace only handles event emission.
 
Let me know if this needs change.

Reply:
That approach sounds good to me.